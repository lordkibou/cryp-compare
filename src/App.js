import { BrowserRouter , Routes, Route }
  from 'react-router-dom';
import { loadCryptoMarketsPrices, loadCryptoInfo } from "./clientFunctions/loadCryptoMarketsPrices";
import HomePage from './scenes/homePage';
import MarketsComparator from './scenes/marketsComparator';
import './index.css';

document.addEventListener('DOMContentLoaded', async function () {
  // First page: 0, with 50 cryptos
  /*
  SE LLAMAN DENTRO DEL USE EFFECT
  await loadCryptoMarketsPrices( "DOGE" , function (res , err) {
    if (err) {
      console.log(err)
      return
    }
    console.log(res)
  });
  await loadCryptoInfo(0, function (res, err) {
    if (err) {
      console.log(err)
      return
    }
    console.log(res)
  })
  */
})
function App() {
  return (
    <div className="App">
      <BrowserRouter>
        <Routes>
          <Route path='/' element={<HomePage/>} />
          <Route path='/markets' element={<MarketsComparator/>}/>
        </Routes>
      </BrowserRouter>
    </div>
  );
}

export default App;
