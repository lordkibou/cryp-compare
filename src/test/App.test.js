import renderer from 'react-test-renderer';
import App from '../App'
//If the change is expected you can invoke Jest with jest -u to overwrite the existing snapshot.

it('Testing App.js', () => {
  const component = renderer.create(
    <App/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});