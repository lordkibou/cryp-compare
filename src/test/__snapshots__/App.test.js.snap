// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App.js 1`] = `
<div
  className="App"
>
  <div
    className="loading-container"
  >
    <span
      style={
        Object {
          "display": "inherit",
        }
      }
    >
      <span
        style={
          Object {
            "animation": "react-spinners-BeatLoader-beat 0.7s 0s infinite linear",
            "animationFillMode": "both",
            "backgroundColor": "#007bff",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": "15px",
            "margin": "2px",
            "width": "15px",
          }
        }
      />
      <span
        style={
          Object {
            "animation": "react-spinners-BeatLoader-beat 0.7s 0.35s infinite linear",
            "animationFillMode": "both",
            "backgroundColor": "#007bff",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": "15px",
            "margin": "2px",
            "width": "15px",
          }
        }
      />
      <span
        style={
          Object {
            "animation": "react-spinners-BeatLoader-beat 0.7s 0s infinite linear",
            "animationFillMode": "both",
            "backgroundColor": "#007bff",
            "borderRadius": "100%",
            "display": "inline-block",
            "height": "15px",
            "margin": "2px",
            "width": "15px",
          }
        }
      />
    </span>
  </div>
</div>
`;
